### DISCUSSION TITLE
How do you want to handle links in translations

### DISCUSSION THREAD

**Discussion by cavasinf** [kevinpapst/TablerBundle#54] [Q&A] [answered]
It is common to have link inside a translation.

Multiple ways to handle it:
1. Raw code
https://stackoverflow.com/questions/11545459/adding-a-link-inside-translated-content-in-twig-template

2. External packages (markdown)
https://twig.symfony.com/doc/2.x/filters/markdown_to_html.html

3. Custom Twig Filter
https://larsgraubner.com/blog/symfony/translation-links/

What is the better option for you @kevinpapst?

**Comment by kevinpapst** [comment, marked as answer]
I don't like any of these options (but 1 & 2 definitely not).

Why not 
```twig
{{ 'foo %link%'|trans({ '%link%': '<a href="https://.../discussions/54">' ~ ('Test'|trans) ~ '</a>' }, 'messages')|raw }}
```
?
Another option is to rephrase your translation and simply append the URL to the end of the translation.
```
{{ 'foo %link%'|trans }}: <a href="https://github.com/kevinpapst/TablerBundle/discussions/54">Test</a>
```

**Reply by cavasinf** [reply]
Hmmm, how do you want to translate the text `Test` inside the link with that solution ?
`<a href="https://github.com/kevinpapst/TablerBundle/discussions/54">!!!!!!Test!!!!!!</a>`

**Reply by kevinpapst** [reply]
I fixed the copy & paste error in case 1 and added an example to translate the link content

**Reply by cavasinf** [reply]
I know that french VS English, words order can be reversed.
Like verb is before or after in certain case for example.

In our case if we only do simple trads, we can split it in two separate transaltion phrase:
- one for the "big" phrase
- another one for the link label

Is that good for you ?

**Reply by kevinpapst** [reply]
I had that case in a password email once where you had to click the one-time link and I made it two translations.
One for the large text above and then one button in the middle that used the second translation. 
Sometimes a reorganization of the frontend helps in finding a solution.

But in short: I'd like to see it as simple as possible! 😄 so yes, that sounds good
